hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/onset/OnsetCheckerTest.java

```diff
@@ -0,0 +1,508 @@
+                        .setTransactionValidDuration(ONE_MINUTE)
+                        .build();
+
+        // then
+        assertThatThrownBy(() -> checker.checkTransactionBody(txBody))
+                .isInstanceOf(PreCheckException.class)
+                .hasFieldOrPropertyWithValue("responseCode", DUPLICATE_TRANSACTION);
+    }
+
+    @Test
+    void testCheckTransactionBodyWithInvalidFeeFails() {
+        // given
+        final var txBody =
+                TransactionBody.newBuilder()
+                        .setTransactionID(transactionID)
+                        .setTransactionValidDuration(ONE_MINUTE)
+                        .setTransactionFee(-1L)
+                        .build();
+
+        // then
+        assertThatThrownBy(() -> checker.checkTransactionBody(txBody))
+                .isInstanceOf(PreCheckException.class)
+                .hasFieldOrPropertyWithValue("responseCode", INSUFFICIENT_TX_FEE);
+    }
+
+    @Test
+    void testCheckTransactionBodyWithInvalidAccountNumFails() {
+        // given
+        final var payerId = AccountID.newBuilder().setAccountNum(0L).build();
+        final var now = Timestamp.newBuilder().setSeconds(Instant.now().getEpochSecond()).build();
+        transactionID =
+                TransactionID.newBuilder()
+                        .setAccountID(payerId)
+                        .setTransactionValidStart(now)
+                        .build();
+        final var txBody =
+                TransactionBody.newBuilder()
+                        .setTransactionID(transactionID)
+                        .setTransactionValidDuration(ONE_MINUTE)
+                        .build();
+
+        // then
+        assertThatThrownBy(() -> checker.checkTransactionBody(txBody))
+                .isInstanceOf(PreCheckException.class)
+                .hasFieldOrPropertyWithValue("responseCode", PAYER_ACCOUNT_NOT_FOUND);
+    }
+
+    @Test
+    void testCheckTransactionBodyWithInvalidRealmNumFails() {
+        // given
+        final var payerId = AccountID.newBuilder().setAccountNum(1L).setRealmNum(-1L).build();
+        final var now = Timestamp.newBuilder().setSeconds(Instant.now().getEpochSecond()).build();
+        transactionID =
+                TransactionID.newBuilder()
+                        .setAccountID(payerId)
+                        .setTransactionValidStart(now)
+                        .build();
+        final var txBody =
+                TransactionBody.newBuilder()
+                        .setTransactionID(transactionID)
+                        .setTransactionValidDuration(ONE_MINUTE)
+                        .build();
+
+        // then
+        assertThatThrownBy(() -> checker.checkTransactionBody(txBody))
+                .isInstanceOf(PreCheckException.class)
+                .hasFieldOrPropertyWithValue("responseCode", PAYER_ACCOUNT_NOT_FOUND);
+    }
+
+    @Test
+    void testCheckTransactionBodyWithInvalidAccountFails() {
```

Comment:
Is this method testing anything substantially different than `testCheckTransactionBodyWithInvalidAccountNumFails`?

Reply:
Yes, it tests the check with an invalid `shardNum`. Apparently I forgot to rename it. Fixed it.